### ISSUE TITLE
Sample and Illumination None behaviour

### ISSUE THREAD

**Issue by pierrethibault** [ptycho/ptypy#191]
Sub-parameters for `illumination` and `sample` are announced as `None` by default in the docstrings but the actual defaults are not `None`. As a consequence it is currently essential to pass explicitly `illumination.diversity=None`, `sample.process=None`, etc. This is not the expected behaviour.

**Comment by aaron-parsons** [comment]
Yup. It turns out that #203 was actually this. because process wasn't set to None explicitly (even though the default was None), it tried to process the stxm data and screwed up the result:-)
I'm off tomorrow, but might take a look this evening.

At least we can carry on.

**Comment by aaron-parsons** [comment]
Ok. I think that the neatest way to do this is in Param.update. 

Not something I want to touch in hurry so will look again on Monday.

**Comment by bjoernenders** [comment]
This needs to be reviewed if that behavior is still there.
